### PR TITLE
Patch missing functionality needed for app in tendermint v0.35

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

When upgrading celestia-core, I missed some basic functionality that is needed to update the app.

This PR:
- Apply the malleated tx mempool fix to the prioritized mempool (v1) and not just the old mempool
- Pass the entire block data during `ProcessProposal`

### CODE REVIEW

**Comment by evan-forbes** [comment]
one sec, let me fix the git history

File: internal/mempool/v1/mempool.go [hunk 1]
```diff
@@ -461,6 +462,12 @@ func (txmp *TxMempool) Update(
 		// remove the committed transaction from the transaction store and indexes
 		if wtx := txmp.txStore.GetTxByHash(tx.Key()); wtx != nil {
 			txmp.removeTx(wtx, false)
+		} else if originalHash, _, isMalleated := types.UnwrapMalleatedTx(tx); isMalleated {
+			var origianlKey [sha256.Size]byte
+			copy(origianlKey[:], originalHash)
```
Comment (liamsi) [inline internal/mempool/v1/mempool.go:467]:
```suggestion
			var originalKey [sha256.Size]byte
			copy(originalKey[:], originalHash)
```

File: internal/mempool/v1/mempool.go [hunk 2]
```diff
@@ -461,6 +462,12 @@ func (txmp *TxMempool) Update(
...
+			if wtx := txmp.txStore.GetTxByHash(origianlKey); wtx != nil {
```
Comment (liamsi) [inline internal/mempool/v1/mempool.go:468]:
```suggestion
			if wtx := txmp.txStore.GetTxByHash(originalKey); wtx != nil {
```

**Review by adlerjohn — COMMENTED** [review]
CI failing. Proto breaking is unexpected too.

**Review by adlerjohn — APPROVED** [review]

**Review by liamsi — APPROVED** [review]

**Comment by evan-forbes** [comment]
> CI failing. Proto breaking is unexpected too.

unfortunately, there are quite a few tests in v0.35 that are flaky atm. I'm also still unsure why the proto is still breaking. We did the same changes as #716, but its still breaking.

I'll merge for now, as this is blocking, but we should still investigate further. Worst case scenario we can remove the proto breakage test, but I would prefer to find out why obvi

**Pull request merged**